src/overlays/actors/ovl_En_Ani/z_en_ani.c

```diff
@@ -28,62 +44,314 @@ const ActorInit En_Ani_InitVars = {
+    } else {
+        this->blinkFunc = EnAni_WaitForEyeOpen;
+        this->blinkTimer = Rand_S16Offset(20, 20);
+    }
+}
+
+void EnAni_WaitForEyeOpen(EnAni* this) {
+    if (this->blinkTimer > 0) {
+        this->blinkTimer--;
+    } else if (this->blinkState > 0) {
+        this->blinkState--;
+    } else {
+        this->blinkFunc = EnAni_WaitForEyeClose;
+        this->blinkTimer = Rand_S16Offset(10, 10);
+    }
+}
+
+void EnAni_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnAni* this = THIS;
+    s32 pad;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    ActorShape_Init(&this->actor.shape, 0.0f, func_800B3FC0, 24.0f);
+    SkelAnime_InitFlex(globalCtx, &this->skelAnime, &gAniSkeleton, &gAniStandingNormalAnim, this->jointTable,
+                       this->morphTable, ANI_LIMB_COUNT);
+    Animation_PlayOnce(&this->skelAnime, &gAniStandingNormalAnim);
+    Collider_InitAndSetCylinder(globalCtx, &this->collider1, &this->actor, &sCylinderInit);
+    Collider_InitAndSetCylinder(globalCtx, &this->collider2, &this->actor, &sCylinderInit);
+    Collider_UpdateCylinder(&this->actor, &this->collider2);
+    this->actor.colChkInfo.mass = MASS_IMMOVABLE;
+    this->stateFlags = ANI_STATE_STANDING;
+    this->unusedShort = 0;
+    this->treeReachTimer = 0;
+    this->blinkFunc = EnAni_DefaultBlink;
+
+    if (GET_ANI_TYPE(thisx) == ANI_TYPE_TREE_HANGING) {
+        Animation_Change(&this->skelAnime, &gAniTreeHangingAnim, 1.0f, 0.0f,
+                         Animation_GetLastFrame(&gAniTreeHangingAnim), 2, 0.0f);
+        this->actionFunc = EnAni_TreeHanging;
+        this->actor.velocity.y = 0.0f;
+        this->actor.minVelocityY = 0.0f;
+        this->actor.gravity = 0.0f;
+        this->actor.flags |= 0x10;
+        this->stateFlags |= ANI_STATE_CLIMBING;
+        gSaveContext.eventInf[1] &= (u8)~0x10;
+
+    } else { // ANI_TYPE_UNUSED_STANDING
+        // for some reason standing he has a large collider
+        // possibly he was meant to be a blocking npc like bomber
+        this->collider2.dim.radius = 60;
+        this->actionFunc = EnAni_IdleStandingUnused;
+        this->actor.velocity.y = -25.0f;
+        this->actor.minVelocityY = -25.0f;
+        this->actor.gravity = -5.0f;
+    }
+}
+
+void EnAni_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnAni* this = THIS;
+    Collider_DestroyCylinder(globalCtx, &this->collider1);
+    Collider_DestroyCylinder(globalCtx, &this->collider2);
+}
+
+void EnAni_SetText(EnAni* this, GlobalContext* globalCtx, u16 textId) {
+    s16 diffAngle = this->actor.yawTowardsPlayer - this->actor.shape.rot.y;
+
+    this->actor.textId = textId;
+    if ((this->stateFlags & ANI_STATE_SUFFERING) != 0 || ABS_ALT(diffAngle) < 0x4301) {
+        if (this->actor.xzDistToPlayer < 100.0f) {
+            func_800B8614(&this->actor, globalCtx, 120.0f);
+        }
+    }
+}
+
+/**
+ * just stands there staring at you,
+ * this is NOT him as a shopkeeper, that is EnOsn using his object
```

Comment:
Ossan? Isn't Osn the Happy Mask salesman? Either way, that comment is probably more appropriate to the top of the file.